File: Challenge2/JavaScript/Challenge2.js
function Fibonacci() {
    var firstNumber = 0;
    var secondNumber = 1;
    var totalNumber = 0;
    var result = 0;

    while (result < 4000000) {
        firstNumber = secondNumber;
        secondNumber = result;
        result = firstNumber + secondNumber;
        if (result % 2 == 0) {
            totalNumber += result;
        }
    }
    return totalNumber;
}

console.log(Fibonacci());

File: Challenge2/Java/Challenge2.java
public class Challenge2 {
	
	public static void main(String[] args){
		System.out.println(Fibonacci());
	}
		
		public static int Fibonacci(){
			int firstNumber = 0;
			int secondNumber = 1;
			int nextNumber = 0;
			int totalNumber = 0;
			
			while (nextNumber < 4000000){
				nextNumber = firstNumber + secondNumber;
				firstNumber = secondNumber;
				secondNumber = nextNumber;
				if	(nextNumber % 2 == 0){
					totalNumber += nextNumber;
				}}
			return totalNumber;
		}
}